gcloud-java-storage/src/main/java/com/google/gcloud/storage/Storage.java

```diff
@@ -555,27 +555,37 @@ public Builder sourceOptions(Iterable<BlobSourceOption> options) {
        *
        * @return the builder.
        */
-      public Builder target(BlobInfo target) {
-        this.target = target;
+      public Builder target(BlobId target) {
+        this.target = BlobInfo.builder(target).build();
         return this;
       }
 
       /**
-       * Sets blob's target options.
+       * Sets the copy target and target options. Copied blob metadata is set to {@code target}.
```

Comment:
I think it would be nice to make the distinction between providing `BlobId` and `BlobInfo` suggesting that the latter is used to override the source information (such as content-type, content-language and metadata) and in that case content-type is a required field.